docs/state.md

```diff
@@ -186,3 +186,32 @@ if err != nil {
 gcsClient, err := storage.New(httpClient)
 
 ```
+
+## Vault (vault://)
+
+The goal of the vault store is to be a safe storage for the kops keys and secrets store. It will not work to use this as a kops registry/config store. Among other things, etcd-manager is not able to read VFS control file from vault. Vault can also not be used as backend for etcd backups.
+
+The vault store uses IAM auth to authenticate against the vault server and expects the vault auth plugin to be mounted on /aws. Vault client locally will expect VAULT_TOKEN to be set.
```

Comment:
This isn't entirely accurate.
On a server, it uses AWS auth to authenticate to Vault server.
When running kops locally, you naturally cannot use Vault server, and kops expects `VAULT_TOKEN` env variable to be set. Any authentication method using `vault` client will generate a token, so the auth method doesn't really matter.

Reply:
I found the sentence confusing and could not quite figure out what it was trying to convey. The term "vault client locally" is ambiguous/unclear. It seems the intent is to use `VAULT_TOKEN` when running the kops CLI and IAM during nodeup. The code simply keys off of the presence/absence of `VAULT_TOKEN`.

Getting a token to nodeup would require some sort of customization, but I could see a site injecting a script to use a (non-AWS) instance credential to obtain one.

It's not immediately obvious to me why one couldn't use IAM when running the kops CLI, aside from the code's current expectation for the plugin to be at a hardcoded `/aws`.

Perhaps it would be useful to separate the descriptions of mechanism and intent.

Reply:
There was also confusion about whether VAULT_TOKEN was intended for use in integration tests against an on-box vault server.

Reply:
Tried to make this a bit more clear now.